homeassistant/components/scrape/config_flow.py

```diff
@@ -73,32 +77,79 @@
         TextSelectorConfig(type=TextSelectorType.PASSWORD)
     ),
     vol.Optional(CONF_HEADERS): ObjectSelector(),
-    vol.Optional(CONF_UNIT_OF_MEASUREMENT): TextSelector(),
+    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): BooleanSelector(),
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): NumberSelector(
+        NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+    ),
+}
+
+SENSOR_SETUP_OPT = {
+    vol.Required(CONF_SELECT): TextSelector(),
+    vol.Optional(CONF_INDEX, default=0): NumberSelector(
+        NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+    ),
+    vol.Optional(CONF_ATTRIBUTE): TextSelector(),
+    vol.Optional(CONF_VALUE_TEMPLATE): TemplateSelector(),
     vol.Optional(CONF_DEVICE_CLASS): SelectSelector(
         SelectSelectorConfig(
-            options=[e.value for e in SensorDeviceClass],
+            options=[cls.value for cls in SensorDeviceClass],
             mode=SelectSelectorMode.DROPDOWN,
         )
     ),
     vol.Optional(CONF_STATE_CLASS): SelectSelector(
         SelectSelectorConfig(
-            options=[e.value for e in SensorStateClass],
+            options=[cls.value for cls in SensorStateClass],
             mode=SelectSelectorMode.DROPDOWN,
         )
     ),
-    vol.Optional(CONF_VALUE_TEMPLATE): TemplateSelector(),
-    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): BooleanSelector(),
+    vol.Optional(CONF_UNIT_OF_MEASUREMENT): SelectSelector(
+        SelectSelectorConfig(
+            options=[cls.value for cls in UnitOfTemperature],
+            custom_value=True,
+            mode=SelectSelectorMode.DROPDOWN,
+        )
+    ),
+}
+
+SENSOR_SETUP = {
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): TextSelector(),
 }
 
-DATA_SCHEMA = vol.Schema({**SCHEMA_SETUP, **SCHEMA_OPT})
-DATA_SCHEMA_OPT = vol.Schema({**SCHEMA_OPT})
+
+def validate_rest_setup(user_input: dict[str, Any]) -> dict[str, Any]:
+    """Validate rest setup."""
+    hass = async_get_hass()
+    rest_config: dict[str, Any] = COMBINED_SCHEMA(user_input)
+    try:
+        rest = create_rest_data_from_config(hass, rest_config)
+        asyncio.run_coroutine_threadsafe(rest.async_update(), hass.loop)
```

Comment:
I think you still need to wait for the result:
```suggestion
        asyncio.run_coroutine_threadsafe(rest.async_update(), hass.loop).result()
```

Reply:
Note => I don't think we can do this.
Maybe we have to remove the line altogether, ensure only that create_rest_data_from_config succeeds, and validate `async_update` in a follow-up PR.

Reply:
was about to write that doesn't work. Will remove the line for now.